### PR TITLE
Iris multiview/aic 5533 v2

### PR DESCRIPTION
**Pull request by ShevchenkoRostyslav**
# MultiFrame Templates Aggregation pipeline

## PR description

This PR Implements a new pipeline for aggregating multiple iris templates from the same user to enhance recognition performance.

### Core Architecture Refactoring
- **BasePipeline abstraction**: New generic base class extracting shared pipeline logic for better code reuse and maintainability
- **IrisPipeline refactoring**: Migrated existing pipeline to inherit from BasePipeline, significantly reducing code duplication

### New Multiframe Aggregation Pipeline
- **MultiframeAggregationPipeline**: Specialized pipeline for template aggregation with configurable environments and output builders
- **Complete pipeline configuration**: YAML-based configuration defining the full iris processing workflow for multiframe scenarios
- **Majority Vote Algorithm**: Advanced aggregation using selective bits fusion, consistency analysis, and fragile bit handling

### Template Validation & Compatibility
- **AreTemplatesAggregationCompatible validator**: Ensures templates have matching dimensions, iris code versions, and wavelet counts before aggregation

### Output Builders & Environment Support
- New output builders for different aggregation result formats
- Environment-based pipeline configuration supporting multiple output types (ORB, simple aggregation)

### Configurable Parameters
- **Consistency threshold**: Controls bit consistency requirements across templates
- **Mask threshold**: Determines valid mask bit criteria in combined templates  
- **Fragile bit threshold**: Configures fragile bit identification and usage
- **Weight-based reliability**: Generates weight matrices reflecting bit reliability across templates

### Backward Compatibility
- Existing `IrisPipeline` functionality fully preserved through inheritance
- Breaking changes to current API or pipeline configurations:
- - The names of the `pipeline_output_builder` provided into the `IRISPipeline` environments have been changed;
- - Non-essential APIs have been made private: `instanciate_nodes` -> `_instanciate_nodes` etc.

### Comprehensive Test Suite
- **E2E tests**: Complete end-to-end testing for the multiframe aggregation pipeline
- **Unit tests**: Thorough coverage of majority vote algorithm and pipeline functionality  

### Test Categories
- Pipeline execution and configuration loading
- Template aggregation algorithms and edge cases
- Compatibility validation and error conditions
- Output builder functionality and format verification

## Limitations

1. See **Backward Compatibility** chapter
2. Templates aggregation logic implements a majority vote for the iris bits across the N provided templates:
```[python]
# Count votes and valid masks
        for i in range(num_templates):
            # Count votes for iris code bits (1s)
            vote_counts += iris_codes[i] * mask_codes[i]

            # Count valid mask bits
            valid_mask_counts += mask_codes[i]

        # Calculate the fraction of votes for each bit
        vote_fractions = vote_counts / np.maximum(valid_mask_counts, 1)

        # Create combined iris code using majority voting
        combined_iris_code = (vote_fractions > 0.5).astype(bool)
```
This means that if a certain bit has a non-zero iris_code but is masked out, its "vote" for this bit is not counted. A known edge case is that if one combines N identical templates, the combined template is NOT going to be identical to them, but instead equal to `iris_code * mask_code` of the input template.

## Type

- [x] Feature
- [x] Refactoring
- [ ] Bugfix
- [ ] DevOps
- [ ] Testing

## Checklist

- [x] I've made sure that my code works as expected by writing unit tests.
- [x] I've checked if my code doesn't generate warnings or errors.
- [x] I've performed a self-review of my code.
- [x] I've made sure that my code follows the style guidelines of the project.
- [x] I've commented hard-to-understand parts of my code.
- [x] I've made appropriate changes in the documentation.

### CODE REVIEW

File: src/iris/pipelines/iris_pipeline.py [hunk 1]
```diff
@@ -59,6 +63,9 @@ class Parameters(Algorithm.Parameters):
         metadata: PipelineMetadata
         pipeline: List[PipelineNode]
 
+        class Config(ImmutableModel.Config):
+            extra = Extra.ignore
```
Comment (ShevchenkoRostyslav) [inline src/iris/pipelines/iris_pipeline.py:67]: @wiktorlazarski please double check whether this is acceptable for you

File: src/iris/pipelines/iris_pipeline.py [hunk 2]
```diff
@@ -74,244 +81,89 @@ def __init__(
...
+            self.env.error_manager(self.call_trace, ValueError(f"Could not find node {node.name}."))
+            return False
+        else:
+            # For all other errors, use the default error manager
+            self.env.error_manager(self.call_trace, error)
+            return False
 
-        Raises:
-            IRISPipelineError: Raised if a node's inputs are not declared beforehands
+    @classmethod
```
Comment (ShevchenkoRostyslav) [inline src/iris/pipelines/iris_pipeline.py:142]: @wiktorlazarski this is now a `@classmethod` in order to be able to properly propagate class hierarchy, i.e. priv_iris does not need to overwrite this method anymore afaik

**Review by wiktorlazarski — CHANGES_REQUESTED** [review]
Looks good to me! Nice work! Just one minor thing before final approval.

**Review by wiktorlazarski — APPROVED** [review]
